File: hsg9-tinh/2022-2023-thuathienhue/matma/matma.cpp
#include <iostream>

#define input_file "matma.inp"
#define output_file "matma.out"

using namespace std;

int mom_day, mom_month, mom_year;
int dad_day, dad_month, dad_year;

// mật mã cần tìm
int code;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> mom_day >> mom_month >> mom_year;

    cin >> dad_day >> dad_month >> dad_year;
}


// Hàm tìm ước chung lớn nhất
int gcd(int a, int b)
{
    while (b > 0)
    {
        a %= b;
        swap(a, b);
    }

    return a;
}


void process()
{
    int mom_sum = mom_day + mom_month + mom_year;

    int dad_sum = dad_day + dad_month + dad_year;

    int g = gcd(mom_sum, dad_sum);

    code = mom_sum / g * dad_sum;
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << code; 
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2022-2023-thuathienhue/matma/matma.py
import os
import sys

input_file = os.path.join(sys.path[0], 'matma.inp')
output_file = os.path.join(sys.path[0], 'matma.out')

mom_day = mom_month = mom_year = 0
dad_day = dad_month = dad_year = 0

# mật mã cần tìm
code = 0


def input():
    global mom_day, mom_month, mom_year, dad_day, dad_month, dad_year

    with open(input_file, 'r') as f:
        mom_day, mom_month, mom_year = map(int, f.readline().split())
        dad_day, dad_month, dad_year = map(int, f.readline().split())


# Hàm tìm ước chung lớn nhất
def gcd(a, b):
    while b > 0:
        a, b = b, a % b

    return a


def process():
    global mom_day, mom_month, mom_year, dad_day, dad_month, dad_year, code

    mom_sum = mom_day + mom_month + mom_year

    dad_sum = dad_day + dad_month + dad_year

    g = gcd(mom_sum, dad_sum)

    code = mom_sum // g * dad_sum


def output():
    global code

    with open(output_file, 'w') as f:
        f.write(str(code))


if __name__ == '__main__':
    input()
    process()
    output()